File: frontend/src/types/service.ts
export interface Service {
  id: number;
  company_id: number;
  name: string;
  description?: string | null;
  unit_price: number;
  tax_rate_id?: number | null;
  category_id?: number | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface ServiceCreate {
  company_id: number;
  name: string;
  description?: string | null;
  unit_price: number;
  tax_rate_id?: number | null;
  category_id?: number | null;
  is_active?: boolean;
}

export interface ServiceUpdate extends Partial<ServiceCreate> {}

File: backend/app/schemas/service.py
from datetime import datetime
from typing import List

from pydantic import BaseModel


class ServiceCategoryBase(BaseModel):
    name: str
    description: str | None = None


class ServiceCategoryCreate(ServiceCategoryBase):
    company_id: int


class ServiceCategoryRead(ServiceCategoryBase):
    id: int
    company_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class TaxRateBase(BaseModel):
    name: str
    rate_percent: float
    is_default: bool = False


class TaxRateCreate(TaxRateBase):
    company_id: int


class TaxRateRead(TaxRateBase):
    id: int
    company_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ServiceBase(BaseModel):
    name: str
    description: str | None = None
    unit_price: float
    tax_rate_id: int | None = None
    category_id: int | None = None
    is_active: bool = True


class ServiceCreate(ServiceBase):
    company_id: int


class ServiceUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    unit_price: float | None = None
    tax_rate_id: int | None = None
    category_id: int | None = None
    is_active: bool | None = None


class ServiceRead(ServiceBase):
    id: int
    company_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

